app/connectors/IncomeTaxSubscriptionConnector.scala

```diff
@@ -48,6 +49,11 @@ class IncomeTaxSubscriptionConnector @Inject()(appConfig: AppConfig,
     http.DELETE[HttpResponse](deleteURL(reference))
   }
 
+  def deleteSubscriptionDetails[T](reference: String, key: String)
```

Comment:
```suggestion
  def deleteSubscriptionDetails(reference: String, key: String)
```
can remove the generic type as it's not used here